src/cacao.ts

```diff
@@ -78,6 +89,48 @@ export namespace Cacao {
 
     return cacao
   }
+
+  export function verify(cacao: Cacao, options: VerifyOptions = {}): VerificationResult {
```

Comment:
@haardikk21 You either rely on throwing an error, or _always_ return `VerificationResult`. Do not conflate two control flow mechanisms, please. Errors seem easier.